src/main/java/nus/climods/model/module/UserModule.java

```diff
@@ -34,13 +38,38 @@ public UserModule(String moduleCode, Model model) throws CommandException {
         }
 
         listModule = optionalModule.get();
+        //default put semester 1
+        this.semester = SemestersEnum.S1;
+    }
+
+    /**
+     * Creates a UserModule with Semester details
+     * @param moduleCode String for the module code
+     * @throws CommandException if module code is not a valid module in current NUS curriculum or module is not
+     *      offered in specified semester.
+     */
+    public UserModule(String moduleCode, SemestersEnum semester, Model model) throws CommandException {
```

Comment:
A data model like `UserModule` should not be responsible for throwing `CommandException`. All command related errors should be caught by `Command` or `CommandParser`